hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/layout/listener/TenantConfigChangedListener.java

```diff
@@ -0,0 +1,147 @@
+/** 
+ * Copyright (c) 2022 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.common.layout.listener;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Consumer;
+
+import org.eclipse.hawkbit.tenancy.configuration.TenantConfigurationProperties;
+import org.eclipse.hawkbit.ui.common.event.EventTopics;
+import org.eclipse.hawkbit.ui.common.event.TenantConfigChangedEventPayload;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+/**
+ * Event listener for tenant configuration changes
+ */
+public class TenantConfigChangedListener extends TopicEventListener {
+
+    private final String tenantFilter;
+    private final Collection<String> tenantConfigFilter;
+    private final Consumer<TenantConfigChangedEventPayload> onApplicationScope;
+    private final Consumer<TenantConfigChangedEventPayload> onUiScope;
+
+    /**
+     * Constructor for TenantConfigChangedListener
+     *
+     * @param eventBus
+     *            UIEventBus
+     * @param tenantFilter
+     *            filter events related to a specific tenant
+     * @param tenantConfigFilter
+     *            filter the tenant configurations of interest listed in
+     *            {@link TenantConfigurationProperties.TenantConfigurationKey}
+     * @param onApplicationScope
+     *            defines what should happen on an application wide event
+     * @param onUiScope
+     *            defines what should happen on current UI event
+     */
+    TenantConfigChangedListener(final UIEventBus eventBus, final String tenantFilter,
```

Comment:
Incase we need this class, then we need to add javadoc to all the public methods here

Reply:
added